modules/pubmaticBidAdapter.js

```diff
@@ -603,42 +612,85 @@ function _createImpressionObject(bid, conf) {
     bidfloorcur: bid.params.currency ? _parseSlotParam('currency', bid.params.currency) : DEFAULT_CURRENCY
   };
 
-  if (bid.params.hasOwnProperty('video')) {
-    var videoData = bid.params.video;
+  if (bid.hasOwnProperty('mediaTypes')) {
+    for (mediaTypes in bid.mediaTypes) {
+      switch (mediaTypes) {
+        case BANNER:
+          bannerObj = {};
```

Comment:
can we move following banner specific code in a separate function like _createNativeRequest?
smaller functions make code easy to understand and maintain.

Reply:
Ok.